Src/VimWpf/Implementation/MarkGlyph/MarkGlyphTaggerSource.cs

```diff
@@ -119,10 +119,40 @@ private void OnTextBufferChanged(object sender, TextContentChangedEventArgs e)
 
         private void OnHideMarksChanged(object sender, EventArgs e)
         {
-            _hideMarks = _markDisplayUtil.HideMarks;
+            _hideMarks = ExpandHideMarksSetting(_markDisplayUtil.HideMarks);
             UpdateAllMarks();
         }
 
+        HashSet<char> ExpandHideMarksSetting(string setting)
+        {
+            var result = new HashSet<char>();
+            if (setting == "-")
+            {
+                setting = "!-~";
+            }
+            var i = 0;
+            while (i < setting.Length)
+            {
+                var c1 = setting[i];
+                if (i + 2 < setting.Length && setting[i + 1] == '-')
+                {
+                    // Expand range, e.g. 'a-z'.
+                    var c2 = setting[i + 2];
+                    for (var c = c1; c <= c2; c++)

```

Comment:
If the user enters "z-a" we won't add z, - or a to the hide list.

Reply:
We now allow a range to be specified in either order.